### PR TITLE
odfe 1.8 version upgrade

### PR DESCRIPTION
**Pull request by vamshin**
*Issue #, if available:*
https://github.com/opendistro-for-elasticsearch/k-NN/issues/113

*Description of changes:*
ODFE 1.8 support with Elasticsearch oss version 7.7.0

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/test-workflow.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ jobs:
   build:
     strategy:
       matrix:
-        java: [13]
+        java: [14]
```
Comment (jmazanec15) [inline .github/workflows/test-workflow.yml:11]: Need to update in other 2 Github actions as well

File: build.gradle [hunk 2]
```diff
@@ -76,8 +71,37 @@ allprojects {
...
+            fileTree(dir: it)
+        }))
+    }
+}
+test.finalizedBy(project.tasks.jacocoTestReport)
+jacocoTestCoverageVerification {
+    violationRules {
+        rule {
+            limit {
+                minimum = 0.5
```
Comment (jmazanec15) [inline build.gradle:93]: Might be worth adding a comment about test coverage in the ReadMe.

**Review by jmazanec15 — COMMENTED** [review]
Minor comments. Just to confirm, everything worked as expected on MultiNode as well?

File: .github/workflows/test-workflow.yml (hunk 1 above)
Comment (vamshin) [reply .github/workflows/test-workflow.yml:11]: Good catch. Done

**Comment by vamshin** [comment]
> Minor comments. Just to confirm, everything worked as expected on MultiNode as well?

Done sanity with multi nodes. Also ran ```./gradlew :integTest -PnumNodes=2```. Tests passed

File: build.gradle (hunk 2 above)
Comment (vamshin) [reply build.gradle:93]: Not sure if coverage needs to be added to README. Seems ok to me.

Comment (jmazanec15) [reply build.gradle:93]: Yes, probably not necessary.

**Review by jmazanec15 — APPROVED** [review]
LGTM!

**Pull request merged**